ids/node_id.go

```diff
@@ -60,35 +52,60 @@ func (id *NodeID) UnmarshalJSON(b []byte) error {
 	}
 
 	var err error
-	*id, err = NodeIDFromString(str[1:lastIndex])
+	*nodeID, err = NodeIDFromString(str[1:lastIndex])
 	return err
 }
 
-func (id *NodeID) UnmarshalText(text []byte) error {
-	return id.UnmarshalJSON(text)
+func (nodeID NodeID) MarshalText() ([]byte, error) {
+	return []byte(nodeID.String()), nil
+}
+
+func (nodeID *NodeID) UnmarshalText(text []byte) error {
+	return nodeID.UnmarshalJSON(text)
 }
 
-func (id NodeID) Less(other NodeID) bool {
-	return bytes.Compare(id[:], other[:]) == -1
+func (nodeID NodeID) Less(other NodeID) bool {
+	return nodeID < other
 }
 
-// ToNodeID attempt to convert a byte slice into a node id
-func ToNodeID(bytes []byte) (NodeID, error) {
-	nodeID, err := ToShortID(bytes)
-	return NodeID(nodeID), err
+func (nodeID NodeID) Equal(other NodeID) bool {
+	return nodeID == other
+}
+
+func NodeIDFromBytes(bytes []byte) NodeID {
+	return NodeID(string(bytes))
+}
+
+// NodeIDFromShortNodeID attempt to convert a byte slice into a node id
+func NodeIDFromShortNodeID(nodeID ShortNodeID) NodeID {
+	return NodeID(string(nodeID.Bytes()))
 }
 
 func NodeIDFromCert(cert *staking.Certificate) NodeID {
-	return hashing.ComputeHash160Array(
+	bytes := hashing.ComputeHash160Array(
 		hashing.ComputeHash256(cert.Raw),
 	)
+	return NodeID(string(bytes[:]))
+}
+
+func (nodeID NodeID) ToSize(newSize int) NodeID {
+	if len(nodeID) >= newSize {
+		return nodeID // leave unchanged if it's longer than required
+	}
+	pad := make([]byte, newSize-len(nodeID))
+	for idx := range pad {
+		pad[idx] = 0x00
+	}
+	bytes := nodeID.Bytes()
+	bytes = append(bytes, pad...)
+	return NodeIDFromBytes(bytes)
 }
 
 // NodeIDFromString is the inverse of NodeID.String()
 func NodeIDFromString(nodeIDStr string) (NodeID, error) {
```

Comment:
Since NodeID is already a string it feels like we should just get rid of this function

Reply:
Not really. This function picks the serialized version of a NodeID from a string and turns it into bytes which are stored in a string.